lightningd/peer_htlcs.c

```diff
@@ -637,7 +650,68 @@ static struct htlc_accepted_hook_response *
 htlc_accepted_hook_deserialize(const tal_t *ctx, const char *buffer,
 			       const jsmntok_t *toks)
 {
```

Comment:
This API is a bit awkward.  We don't really use `htlc_accepted_continue`; instead we use a NULL response from deserialize() to imply it.

And the struct is really a discriminated union already; I think in this case we should simply open-code those parameters into htlc_accepted_hook_deserialize, which is still awkward but at least explicit:

```C
static enum htlc_accepted_result htlc_accepted_hook_deserialize(const char *buffer, const jsmntok_t *toks,
                                                                /* If accepted */ 
                                                                struct preimage *payment_preimage, 
                                                                /* If rejected */
                                                                enum onion_type *failure_code,
                                                                u8 **channel_update)
```